### PR TITLE
Remove repository-related feature flags, which are no longer used

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Remove`createProjectModal`,`configurationsAndPrebuilds` and `showPrebuildsMenuItem` feature flags, since they are no longer useful.

## Related Issue(s)

Fixes ENT-307

## How to test

1. Look at the application menu and make sure all links to Repositories and Prebuilds are there, while the one for Projects is no longer.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-repo-ff-cleanup</li>
	<li><b>🔗 URL</b> - <a href="https://ft-repo-ff-cleanup.preview.gitpod-dev.com/workspaces" target="_blank">ft-repo-ff-cleanup.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-repo-ff-cleanup-gha.26498</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-repo-ff-cleanup%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/menu/Menu.tsx [hunk 1]
```diff
@@ -133,9 +130,6 @@ type OrgPagesNavProps = {
 };
 const OrgPagesNav: FC<OrgPagesNavProps> = ({ className }) => {
     const location = useLocation();
-    const hasMemberPermission = useHasRolePermission(OrganizationRole.MEMBER);
-    const configurationsEnabled = useHasConfigurationsAndPrebuildsEnabled();
-    const prebuildsInMenu = useFeatureFlag("showPrebuildsMenuItem");
 
     const leftMenu: Entry[] = useMemo(() => {
```
Comment (mustard-mh) [inline components/dashboard/src/menu/Menu.tsx:134]: Do we still need useMemo?

File: components/dashboard/src/repositories/coachmarks/MigrationCoachmark.tsx [hunk 2]
```diff
@@ -22,9 +21,6 @@ type Props = PropsWithChildren<{}>;
 export const ConfigurationsMigrationCoachmark = ({ children }: Props) => {
```
Comment (mustard-mh) [inline components/dashboard/src/repositories/coachmarks/MigrationCoachmark.tsx:21]: It's been 4 monthes, maybe we can remove this component

```
// For the users signing up after our launch of configurations, don't show it
if (user.createdAt && user.createdAt.toDate() > new Date("2/21/2024")) {
```

**Review by mustard-mh — APPROVED** [review]

File: components/dashboard/src/menu/Menu.tsx (hunk 1 above)
Comment (filiptronicek) [reply components/dashboard/src/menu/Menu.tsx]: Removed this, thanks for noticing.

File: components/dashboard/src/repositories/coachmarks/MigrationCoachmark.tsx (hunk 2 above)
Comment (filiptronicek) [reply components/dashboard/src/repositories/coachmarks/MigrationCoachmark.tsx:21]: I think it's worth investigating how many users have not seen the banner yet. I am sure there still could be users returning after a couple of months, and keeping this around doesn't require too much maintenance.

Made ENT-321 to track.

**Pull request merged**